core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1567,24 +1518,104 @@ class KafkaApis(val requestChannel: RequestChannel,
     })
 
     if(shareSessionEpoch == ShareFetchMetadata.FINAL_EPOCH && !shareFetchDataWithPartitionMaxBytes.isEmpty) {
+      isValidRequest = false
       shareFetchResponse = shareFetchRequest.getErrorResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, Errors.INVALID_REQUEST.exception) match {
         case response: ShareFetchResponse => response
         case _ => null
       }
-    } else {
+      shareAcknowledgeResponse = shareFetchRequest.getErrorAcknowledgeResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, Errors.INVALID_REQUEST.exception) match {
+        case response: ShareAcknowledgeResponse => response
+        case _ => null
+      }
+    }
+
+    if(isValidRequest) {
       val newReqMetadata : ShareFetchMetadata = new ShareFetchMetadata(Uuid.fromString(memberId), shareSessionEpoch)
       val shareFetchContext = sharePartitionManager.newContext(groupId, shareFetchData, forgottenTopics, topicNames, newReqMetadata)
```

Comment:
Now you have this context, I don't see how it's used to cause the error codes for unknown member ID or incorrect epoch to be returned.